### PR TITLE
Accepting user cases

### PR DESCRIPTION
**Pull request by MichaelParadis**
Provides the testing for accepting user cases. This closes issue #6.

### CODE REVIEW

**Comment by Hreherch** [comment]
We think I wrote it wrong in the GitHub Issues, so Sarah will be checking this over now. I am as well, but I will not submit a review.

**Comment by Hreherch** [comment]
driverPendingAccepts:
"Get open request is not return any open requests" spelling/grammar could be enhanced for clarity when this test fails (This same message is in driverAccept)
Further clarity could be provided in other messages, you use getOpenRequests but you don't specify that the implementation of getOpenRequests hasn't returned a request that it should have.
We can use the request to see the drivers, and shouldn't have to use the controller (The view can see the model)

driverAcceptedRequests:
We can pull the model directly, instead of using the controller to getConfirmedDriver? (i.e. view can see model, but cannot edit it)

notifyDriver:
good!

File: app/src/main/java/comcmput301f16t01/github/carrier/Driver.java [hunk 1]
```diff
@@ -8,8 +8,12 @@
  * @see User
  */
 public class Driver extends User {
-
```
Comment (kieter) [inline app/src/main/java/comcmput301f16t01/github/carrier/Driver.java]: i want this back

File: app/src/test/java/comcmput301f16t01/github/carrier/AcceptingTests.java [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+    public void driverAccept(){
+        Rider rider = new Rider("Mike");
+        Request request = new Request(rider, new Location(), new Location());
+        RequestController rc = new RequestController();
+        rc.addRequest(request);
+        Driver driver = new Driver("Cole");
+        assertTrue("Get open request is not returning any open requests", rc.getOpenRequests().size() == 1);
+        rc.addDriver(request, driver);
+        assertTrue("The request is being updated when it shouldn't", rc.getOpenRequests().size() == 1);
+        assertTrue("The request has not been marked as accepted by the driver.", rc.getPendingRequests(driver).size() == 1);
```
Comment (sehoven) [inline app/src/test/java/comcmput301f16t01/github/carrier/AcceptingTests.java]: This is confusing to me. We are talking about requests and their status from two different perspectives: the rider and the driver. Based on the function name alone this is not clear to me which one we are talking about so the two tests were ambiguous. Also, "pending" is not a word that is well-defined within our application. This needs to be addressed in the Glossary.

File: app/src/test/java/comcmput301f16t01/github/carrier/AcceptingTests.java [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+
+
+
+
```
Comment (sehoven) [inline app/src/test/java/comcmput301f16t01/github/carrier/AcceptingTests.java]: Please delete lines 28-31.

File: app/src/test/java/comcmput301f16t01/github/carrier/AcceptingTests.java [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+import static org.junit.Assert.*;
+/**
+ * Created by michael on 11/10/16.
+ */
+
+public class AcceptingTests {
+
+    /**
+     * (US 05.01.01) As a driver, I want to accept a request I agree with and accept that offered payment upon completion.
```
Comment (sehoven) [inline app/src/test/java/comcmput301f16t01/github/carrier/AcceptingTests.java:13]: The user case specifies "accept that offered payment upon completion". This is not addressed anywhere in the test.

Comment (sehoven) [reply app/src/test/java/comcmput301f16t01/github/carrier/AcceptingTests.java:13]: We could also just need to define more clearly when the offered payment is accepted and the request completed.

File: app/src/test/java/comcmput301f16t01/github/carrier/AcceptingTests.java [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+        Driver driver1 = new Driver("WASD");
+        Request request1 = new Request(rider, new Location(), new Location());
+        rc.addRequest(request1);
+        rc.addDriver(request1,driver1);
+        assertFalse("Two drivers share pending requests when they shouldn't", rc.getPendingRequests(driver).equals(rc.getPendingRequests(driver1)));
+        assertTrue("The request has not been marked as accepted by the driver.", request.getAcceptedDrivers().contains(driver));
+        assertTrue("Driver is not able to view their pending requests .", rc.getPendingRequests(driver).size() == 1);
+        assertTrue("Request in pending requests is not the same.", rc.getPendingRequests(driver).get(0).equals(request));
+        assertEquals("The request is not being updated to pending", Request.ACCEPTED, request.getStatus());
+
```
Comment (sehoven) [inline app/src/test/java/comcmput301f16t01/github/carrier/AcceptingTests.java]: Please delete line 56.

File: app/src/test/java/comcmput301f16t01/github/carrier/AcceptingTests.java [hunk 6]
```diff
@@ -0,0 +1,91 @@
...
+        Driver driver = new Driver("Kevin");
+        Request request = new Request(rider, new Location(), new Location());
+        RequestController rc = new RequestController();
+        rc.addRequest(request);
+        rc.addDriver(request, driver);
+        rc.acceptDriver(request, driver);
+        assertTrue("Driver has been added to the request as the confirmed driver", request.getConfirmedDriver().equals(driver));
+        assertFalse("The request is still available to be accepted by other drivers", rc.getAvailableRequests().contains(request));
+        assertEquals("The request is not being updated to confirmed", Request.CONFIRMED, request.getStatus());
+
```
Comment (sehoven) [inline app/src/test/java/comcmput301f16t01/github/carrier/AcceptingTests.java]: Please delete line 73.

File: app/src/test/java/comcmput301f16t01/github/carrier/AcceptingTests.java [hunk 7]
```diff
@@ -0,0 +1,91 @@
...
+        Rider rider = new Rider("Josh");
+        Driver driver = new Driver("Kevin");
+        Request request = new Request(rider, new Location(), new Location());
+        RequestController rc = new RequestController();
+        rc.addRequest(request);
+        rc.addDriver(request,driver);
+        rc.acceptDriver(request,driver);
+        assertTrue("Driver is not being notified about accepted ride offers.", driver.hasNotifications());
+
+
```
Comment (sehoven) [inline app/src/test/java/comcmput301f16t01/github/carrier/AcceptingTests.java]: Please delete lines 88-89.

File: app/src/test/java/comcmput301f16t01/github/carrier/AcceptingTests.java [hunk 8]
```diff
@@ -0,0 +1,91 @@
...
+        assertTrue("The request is being updated when it shouldn't", rc.getOpenRequests().size() == 1);
+        assertTrue("The request has not been marked as accepted by the driver.", rc.getPendingRequests(driver).size() == 1);
+
+
+
+
+    }
+
+    /**
+     * (US 05.02.01) As a driver, I want to view a list of things I have accepted that are pending, each request with its description, and locations.
```
Comment (sehoven) [inline app/src/test/java/comcmput301f16t01/github/carrier/AcceptingTests.java:38]: It is never addressed that "each request with its description, and locations".

File: app/src/test/java/comcmput301f16t01/github/carrier/AcceptingTests.java [hunk 9]
```diff
@@ -0,0 +1,91 @@
...
+    public void driverAcceptedRequests(){
+        Rider rider = new Rider("Josh");
+        Driver driver = new Driver("Kevin");
+        Request request = new Request(rider, new Location(), new Location());
+        RequestController rc = new RequestController();
+        rc.addRequest(request);
+        rc.addDriver(request, driver);
+        rc.acceptDriver(request, driver);
+        assertTrue("Driver has been added to the request as the confirmed driver", request.getConfirmedDriver().equals(driver));
+        assertFalse("The request is still available to be accepted by other drivers", rc.getAvailableRequests().contains(request));
```
Comment (sehoven) [inline app/src/test/java/comcmput301f16t01/github/carrier/AcceptingTests.java:92]: I kind of understand what "Available Request" does for us but how is this represented in the Request class? I think we need to define this better. I feel like it is too ambiguous.

File: app/src/test/java/comcmput301f16t01/github/carrier/AcceptingTests.java [hunk 10]
```diff
@@ -0,0 +1,91 @@
...
+     * (US 05.03.01) As a driver, I want to see if my acceptance was accepted.
+     */
+    @Test
+    public void driverAcceptedRequests(){
+        Rider rider = new Rider("Josh");
+        Driver driver = new Driver("Kevin");
+        Request request = new Request(rider, new Location(), new Location());
+        RequestController rc = new RequestController();
+        rc.addRequest(request);
+        rc.addDriver(request, driver);
```
Comment (sehoven) [inline app/src/test/java/comcmput301f16t01/github/carrier/AcceptingTests.java:85]: Could we add a test under line 68 to confirm that it IS available to other drivers? (inverse of line 71) Or is this outside the scope of this test case?

**Review by sehoven — CHANGES_REQUESTED** [review]
Please fix the formatting to remove blank lines where necessary.

Some points in the user cases are not addressed. Please address these or give a reason why they are not a concern for the unit tests.

File: app/src/test/java/comcmput301f16t01/github/carrier/AcceptingTests.java (hunk 2 above)
Comment (MichaelParadis) [reply app/src/test/java/comcmput301f16t01/github/carrier/AcceptingTests.java]: Should be addressed.

File: app/src/test/java/comcmput301f16t01/github/carrier/AcceptingTests.java (hunk 3 above)
Comment (MichaelParadis) [reply app/src/test/java/comcmput301f16t01/github/carrier/AcceptingTests.java]: Done

File: app/src/test/java/comcmput301f16t01/github/carrier/AcceptingTests.java (hunk 8 above)
Comment (MichaelParadis) [reply app/src/test/java/comcmput301f16t01/github/carrier/AcceptingTests.java:38]: Done

File: app/src/test/java/comcmput301f16t01/github/carrier/AcceptingTests.java (hunk 5 above)
Comment (MichaelParadis) [reply app/src/test/java/comcmput301f16t01/github/carrier/AcceptingTests.java]: Done

File: app/src/test/java/comcmput301f16t01/github/carrier/AcceptingTests.java (hunk 7 above)
Comment (MichaelParadis) [reply app/src/test/java/comcmput301f16t01/github/carrier/AcceptingTests.java]: Done

File: app/src/test/java/comcmput301f16t01/github/carrier/AcceptingTests.java (hunk 6 above)
Comment (MichaelParadis) [reply app/src/test/java/comcmput301f16t01/github/carrier/AcceptingTests.java]: Done

File: app/src/test/java/comcmput301f16t01/github/carrier/AcceptingTests.java (hunk 9 above)
Comment (sehoven) [reply app/src/test/java/comcmput301f16t01/github/carrier/AcceptingTests.java:92]: This has been addressed in the glossary.

**Review by sehoven — APPROVED** [review]

**Pull request merged**